numpy/core/code_generators/ufunc_docstrings.py

```diff
@@ -2528,6 +2534,46 @@ def add_newdoc(place, name, doc):
 
     """)
 
+add_newdoc('numpy.core.umath', 'clip',
+    """
+    Clip (limit) the values in an array.
+
+    Given an interval, values outside the interval are clipped to
+    the interval edges.  For example, if an interval of ``[0, 1]``
+    is specified, values smaller than 0 become 0, and values larger
+    than 1 become 1.
+
+    Equivalent to but faster than ``np.maximum(a_min, np.minimum(a, a_max))``.
```

Comment:
The [updated property-based test](https://github.com/tylerjereddy/numpy-property/blob/master/tests/test_clip.py#L12) still doesn't agree that this condition is enforced on master or this feature branch.

On feature branch: `Falsifying example: test_clip_scalar_ints(arr=array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), amin=0, amax=-9223372036854775809)`

On master branch: `Falsifying example: test_clip_scalar_ints(arr=array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0]), amin=0, amax=-1)`

Reply:
This was because `npy_ObjectClip` had the operations & operands jumbled around, so that it wasn't a true match to `np.maximum(amin, np.minimum(arr, amax))`.

I've added a unit test for the case used by `hypothesis` along with the patch to fix this.